apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -165,7 +165,7 @@ init_worker(tcp, Pid, Upstreams, Request) ->
 
 -spec(udp_worker(pid(), upstream(), binary()) -> ok).
 udp_worker(Pid, Upstream = {IP, Port}, Request) ->
-    StartTime = erlang:system_time(millisecond),
+    StartTime = erlang:monotonic_time(millisecond),
     MonRef = erlang:monitor(process, Pid),
     Opts = [{reuseaddr, true}, {active, once}, binary],
     {ok, Socket} = gen_udp:open(0, Opts),
```

Comment:
Should will wrap UDP in a `case` as well?

Reply:
Done. I've also wrapped all `send/2` calls.